includes/class-wc-payment-information.php

```diff
@@ -0,0 +1,200 @@
+<?php
+/**
+ * Class WC_Payment_Information
+ *
+ * @package WooCommerce\Payments
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Mostly a wrapper containing information on a single payment.
+ */
+class WC_Payment_Information {
+	/**
+	 * The ID of the payment method used for this payment.
+	 *
+	 * @var string
+	 */
+	private $payment_method;
+
+	/**
+	 * The payment token used for this payment.
+	 *
+	 * @var WC_Payment_Token/NULL
+	 */
+	private $token;
+
+	/**
+	 * Indicates whether this payment was made using a saved card.
+	 *
+	 * @var bool
+	 */
+	private $is_saved_method;
+
+	/**
+	 * Indicates whether the payment is merchant-initiated (true) or customer-initiated (false).
+	 *
+	 * @var bool
+	 */
+	private $off_session;
+
+	/**
+	 * Indicates whether this is a one-off payment (false) or the first installment of a recurring payment (true).
+	 *
+	 * @var bool
+	 */
+	private $is_recurring;
+
+	/**
+	 * Payment information constructor.
+	 *
+	 * @param string                $payment_method The ID of the payment method used for this payment.
+	 * @param WC_Payment_Token/NULL $token The payment token used for this payment.
+	 * @param array                 $is_saved_method Indicates whether this payment was made using a saved card.
+	 * @param bool                  $is_recurring Indicates whether this is a one-off payment (false) or the first installment of a recurring payment (true).
+	 * @param bool                  $off_session Indicates whether the payment is merchant-initiated (true) or customer-initiated (false).
+	 *
+	 * @throws Exception - If no payment method is found in the provided request.
+	 */
+	public function __construct( $payment_method, $token, $is_saved_method = false, $is_recurring = false, $off_session = false ) {
+		$this->payment_method  = $payment_method;
+		$this->token           = $token;
+		$this->is_saved_method = $is_saved_method;
+		$this->is_recurring    = $is_recurring;
+		$this->off_session     = $off_session;
+	}
+
+	/**
+	 * Returns true if payment was made with a saved card, false otherwise.
+	 *
+	 * @return bool True if payment was made with a saved card, false otherwise.
+	 */
+	public function is_using_saved_card() {
+		return $this->is_saved_method;
+	}
+
+	/**
+	 * Returns true if this payment is the first installment of a recurring payment,
+	 * false otherwise.
+	 *
+	 * @return bool True if first installment of recurring payment, false otherwise.
+	 */
+	public function is_first_installment() {
+		return $this->is_recurring;
+	}
```

Comment:
As per https://github.com/Automattic/woocommerce-payments/pull/826#discussion_r478175147 this function probably needs a new name.

Reply:
See https://github.com/Automattic/woocommerce-payments/pull/826#discussion_r478554498